src/state/routing/useRoutingAPITrade.ts

```diff
@@ -25,7 +27,7 @@ export function useRoutingAPITrade<TTradeType extends TradeType>(
   routerPreference: RouterPreference
 ): {
   state: TradeState
-  trade: InterfaceTrade<Currency, Currency, TTradeType> | undefined
+  trade: InterfaceTrade | undefined
```

Comment:
nit: `trade?: InterfaceTrade` personally cleaner and could be used throughout the rest of the PR, but I leave the convention choice up to you!